src/app/modules/OpenshiftStreams/dialogs/CreateInstance/CreateInstanceWithSizes/hooks/useAvailableProvidersAndDefault.ts

```diff
@@ -0,0 +1,136 @@
+    const res = await apisService.getCloudProviderRegions(id);
+
+    if (!res?.data?.items) {
+      return [];
+    }
+
+    const regionsForInstance = res.data.items.filter(
+      (p) => p.enabled && p.capacity.some((c) => c.instance_type === instance_type)
+    );
+
+    return regionsForInstance.map((r): CreateKafkaInstanceWithSizesTypes.RegionInfo => {
+      const max_capacity_reached = r.capacity?.some((c) => c.available_sizes?.length === 0);
+
+      return {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        id: r.id!,
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        displayName: r.display_name!,
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        isDisabled: max_capacity_reached,
+      };
+    });
+  };
+
+  const fetchProviders = async (instance_type: string): Promise<CreateKafkaInstanceWithSizesTypes.Providers> => {
+    try {
+      const apisService = getApi();
+      const res = await apisService.getCloudProviders();
+      const allProviders = res?.data?.items || [];
+      return await Promise.all(
+        allProviders
+          .filter((p) => p.enabled)
+          .map(async (provider): Promise<CreateKafkaInstanceWithSizesTypes.ProviderInfo> => {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const regions = await fetchRegions(provider.id!, instance_type);
+            return {
+              // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+              id: provider.id!,
+              // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+              displayName: provider.display_name!,
+              regions,
+            };
+          })
+      );
+    } catch (e) {
+      console.error('useAvailableProvidersAndDefault', 'fetchProvider', e);
+      return Promise.reject(e);
+    }
+  };
+
+  // TODO - this logic should not exist and be replaced with AMS quota.
+  const fetchUserHasTrialInstance = async (): Promise<boolean> => {
+    try {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const loggedInUser = await getUsername()!;
+      const filter = `owner = ${loggedInUser}`;
+      const apisService = getApi();
+      const res = await apisService.getKafkas('', '', '', filter);
+      if (res.data.items) {
+        return res.data.items.some(
+          (k) => k?.instance_type === InstanceType?.eval || k?.instance_type === InstanceType?.developer
+        );
+      }
+    } catch (e) {
+      console.error('useAvailableProvidersAndDefault', 'fetchUserHasTrialInstance', e);
+    }
+    return false;
+  };
+
+  return async function (): Promise<CreateKafkaInstanceWithSizesTypes.CreateKafkaInitializationData> {
+    try {
+      const quota = await getQuota();
+      const plan = getQuotaType(quota.data);
+      const instance_type = convertQuotaToInstanceType(quota.data);
```

Comment:
```suggestion
      const instanceType = convertQuotaToInstanceType(quota.data);
```

Reply:
ignore some nit/names for now. This code is copied over from other places. I did not write any of this, just reorganized. Will improve it once we have logic in place.